lib/fluent-plugin-s3-arrow/schemas/aws_glue.rb

```diff
@@ -0,0 +1,113 @@
+require 'aws-sdk-glue'
+
+module FluentPluginS3Arrow
+  module Schemas
+    class AWSGlue
+      class Error < RuntimeError; end
+      class ConvertError < Error; end
+      class Field < Struct.new(:name, :type); end
+
+      def initialize(table_name, options={})
+        @table_name = table_name
+        @database_name = options.delete(:database_name) || "default"
+        @catalog_id = options.delete(:catalog_id)
+        @client = Aws::Glue::Client.new(options)
+      end 
+
+      def to_arrow
+        glue_schema = fetch_glue_schema
+        convert_to_arrow_schema(glue_schema)
+      end
+
+      private
+      def fetch_glue_schema
+        glue_table = @client.get_table({
+          catalog_id: @catalog_id,
+          database_name: @database_name,
+          name: @table_name
+        })
+        glue_table.table.storage_descriptor.columns
+      end
+
+      def convert_to_arrow_schema(glue_schema)
+        glue_schema.map do |glue_field|
+          convert_to_arrow_field(glue_field)
+        end
+      end
+
+      def convert_to_arrow_field(glue_field)
+        arrow_field = {name: glue_field.name}
+        case glue_field.type
+        when "boolean", "float", "double"
+          arrow_field[:type] = glue_field.type
+        when "tinyint"
+          arrow_field[:type] = "int8"
+        when "smallint"
+          arrow_field[:type] = "int16"
+        when "int"
+          arrow_field[:type] = "int32"
+        when "bigint"
+          arrow_field[:type] = "int64"
+        when /\Achar/,/\Avarchar/,"string"
+          arrow_field[:type] = "string"
+        when "binary"
+          arrow_field[:type] = "binary"
+        when "date"
+          arrow_field[:type] = "date32"
+        when /\Aarray/
+          arrow_field[:type] = "list"
+          arrow_field[:field] = parse_array(glue_field.type)
+        when /\Astruct/
+          arrow_field[:type] = "struct"
+          arrow_field[:fields] = parse_struct(glue_field.type)
+        else
+          # TODO: Need support for MAP, DECIMAL, TIMESTAMP type.
```

Comment:
If `decimal` and `timestamp` are specified in the schema, json-reader will raise `NotImplemented` error. I don't support it at this time and will add the functionality later.